File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/LABORATOR/lab 2 - home/SortingTask.java
import java.util.Arrays;

public class SortingTask extends Task{
    private int[] numere;
    private Sorter sorter;

    public SortingTask(String id, String descr, int[] numere, SortingStrategy strategy) {
        super(id, descr);
        this.numere = numere;
        sorter = SortingFactory.getInstance().createSorter(strategy);
    }

    @Override
    public void execute() {
        sorter.sort(this.numere);
        System.out.println("Numere sortate: " + Arrays.toString(this.numere));
    }
}

File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/LABORATOR/lab 11 - homework/lab11/lab11/task/SortingTask.cs
namespace lab11;

public class SortingTask:Task
{
    private Task[] _tasks;
    private Strategy strategy;
    private AbstractSort<Task> sorter;

    public SortingTask(string taskID, string description, Task[] tasks, Strategy strategy) : base(taskID, description)
    {
        _tasks = tasks;
        this.strategy = strategy;
    }

    public override void Execute()
    {
        if (strategy == Strategy.BubbleSort)
        {
            sorter=new BubbleSort();
            sorter.Sort(_tasks);
        }
        else
        {   sorter=new QuickSort(); 
            sorter.Sort(_tasks);
        }

        for (int i = 0; i < _tasks.Length; i++)
        {
            Console.WriteLine(_tasks[i]);
        }
    }
}